### PR TITLE
Framework: Get rid of lodash's isNull

### PR DESCRIPTION
**Pull request by tyxla**
There's hardly a more useless lodash method than `isNull`. This PR removes it in favor of the equally simple `=== null`.

#### Changes proposed in this Pull Request

* Framework: Get rid of lodash's `isNull`

#### Testing instructions

* Replacement is straightforward - verify changes make sense and tests pass.

### CODE REVIEW

File: client/components/date-range/index.js [hunk 1]
```diff
@@ -279,8 +278,8 @@ export class DateRange extends Component {
 	 * Converts moment dates to a DateRange
 	 * as required by Day Picker DateUtils
 	 *
-	 * @param  {MomentJSDate} startDate the start date for the range
-	 * @param  {MomentJSDate} endDate   the end date for the range
+	 * @param  {Date} startDate the start date for the range
+	 * @param  {Date} endDate   the end date for the range
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]: If these are meant to be moment date instances then we should keep that accurate:

```suggestion
	 * @param  {import('moment').Moment} startDate the start date for the range
	 * @param  {import('moment').Moment} endDate   the end date for the range
```

File: client/components/date-range/index.js [hunk 2]
```diff
@@ -298,7 +297,7 @@ export class DateRange extends Component {
 	 *
 	 * Dates are only persisted via the commitDates method.
 	 *
-	 * @param  {MomentJSDate} date the newly selected date object
+	 * @param  {Date} date the newly selected date object
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]:
```suggestion
	 * @param  {import('moment').Moment} date the newly selected date object
```

File: client/components/date-range/index.js [hunk 3]
```diff
@@ -588,7 +588,7 @@ export class DateRange extends Component {
 	/**
 	 * Renders the Popover component
 	 *
-	 * @returns {ReactComponent} the Popover component
+	 * @returns {React.Component} the Popover component
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]: These are incorrect, they actually return `JSX.Element`.

```suggestion
	 * @returns {JSX.Element} the Popover component
```

File: client/components/date-range/index.js [hunk 4]
```diff
@@ -627,7 +627,7 @@ export class DateRange extends Component {
 	/**
 	 * Renders the DatePicker component
 	 *
-	 * @returns {ReactComponent} the DatePicker component
+	 * @returns {React.Component} the DatePicker component
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]:
```suggestion
	 * @returns {JSX.Element} the DatePicker component
```

File: client/components/date-range/index.js [hunk 5]
```diff
@@ -679,7 +679,7 @@ export class DateRange extends Component {
 	/**
 	 * Renders the component
 	 *
-	 * @returns {ReactComponent} the DateRange component
+	 * @returns {React.Component} the DateRange component
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]:
```suggestion
	 * @returns {JSX.Element} the DateRange component
```

File: client/components/date-range/index.js [hunk 6]
```diff
@@ -486,10 +485,11 @@ export class DateRange extends Component {
...
 	 *
-	 * @param  {Moment} date             momentJS instance
-	 * @param  {Moment|Date} options.dateFrom the start of the date range
-	 * @param  {Moment|Date} options.dateTo   the end of the date range
-	 * @returns {Moment}                  the date clamped to be within the range
+	 * @param  {Date}  date             momentJS instance
+	 * @param  {Array} options          date range
+	 * @param  {Date}  options.dateFrom the start of the date range
+	 * @param  {Date}  options.dateTo   the end of the date range
+	 * @returns {Date}                  the date clamped to be within the range
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]:
```suggestion
	 * @param  {import('moment').Moment}  date             momentJS instance
	 * @param  {object} options          date range
	 * @param  {import('moment').Moment | Date}  options.dateFrom the start of the date range
	 * @param  {import('moment').Moment | Date}  options.dateTo   the end of the date range
	 * @returns {import('moment').Moment}                  the date clamped to be within the range
```

File: client/components/date-range/index.js [hunk 7]
```diff
@@ -509,7 +509,7 @@ export class DateRange extends Component {
 	 * for display in a text input. Also converts
 	 * to locale appropriate format.
 	 *
-	 * @param  {Date|Moment} date the date for conversion
+	 * @param  {Date} date the date for conversion
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]:
```suggestion
	 * @param  {import('moment').Moment | Date} date the date for conversion
```

File: client/components/date-range/index.js [hunk 8]
```diff
@@ -445,7 +444,7 @@ export class DateRange extends Component {
 	/**
 	 * Converts a moment date to a native JS Date object
 	 *
-	 * @param  {MomentJSDate} momentDate a momentjs date object to convert
+	 * @param  {Date} momentDate a momentjs date object to convert
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]:
```suggestion
	 * @param  {import('moment').Moment} momentDate a momentjs date object to convert
```

File: client/components/date-range/index.js [hunk 9]
```diff
@@ -456,7 +455,7 @@ export class DateRange extends Component {
 	 * Converts a native JS Date object to a MomentJS Date object
 	 *
 	 * @param  {Date} nativeDate date to be converted
-	 * @returns {MomentJSDate}            the converted Date
+	 * @returns {Date}            the converted Date
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]:
```suggestion
	 * @returns {import('moment').Moment}            the converted Date
```

File: client/components/date-range/index.js [hunk 10]
```diff
@@ -466,7 +465,7 @@ export class DateRange extends Component {
 	 * Formats a given date to the appropriate format for the
 	 * current locale
 	 *
-	 * @param  {Date|MomentJSDate} date the date to be converted
+	 * @param  {Date} date the date to be converted
```
Comment (sarayourfriend) [inline client/components/date-range/index.js]:
```suggestion
	 * @param  {import('moment').Moment | Date} date the date to be converted
```

**Review by sarayourfriend — APPROVED** [review]
LGTM. Suggested some nitpicks with the types you updated/changed but otherwise looks good.

File: client/components/date-range/index.js (hunk 1 above)
Comment (tyxla) [reply client/components/date-range/index.js]: Sounds fine to me 👍

File: client/components/date-range/index.js (hunk 3 above)
Comment (tyxla) [reply client/components/date-range/index.js]: Indeed!

**Comment by tyxla** [comment]
Thanks, Sara 🙌

**Pull request merged**